### ISSUE TITLE
 Error when checking target: expected dense_143 to have shape (None, 3) but got array with shape (6210, 1)

### ISSUE THREAD

**Issue by loda717** [keras-team/keras#12378]
Why i got this error message , how can i solve it i try many things but is not working with me 
plz help):

and this my code is  1D Convolutional Neural Network Model 

    model_m = Sequential()
    model_m.add(Conv1D(filters=4, kernel_size=2, activation='relu',input_shape=(n_timesteps,1)))
    model_m.add(LeakyReLU(alpha=0.1))
    model_m.add(MaxPooling1D(2, strides=2))
    model_m.add(Conv1D(4, 5, strides=1))
    model_m.add(LeakyReLU(alpha=0.1))
    model_m.add(MaxPooling1D(2, strides=2))
    model_m.add(Conv1D(10, 4,strides=1))
    model_m.add(LeakyReLU(alpha=0.1))
    model_m.add(MaxPooling1D(2, strides=2))
    model_m.add(Conv1D(10, 4, strides=1))
    model_m.add(LeakyReLU(alpha=0.1))
    model_m.add(MaxPooling1D(2, strides=2))
    model_m.add(Conv1D(15, 4, strides=1))
    model_m.add(LeakyReLU(alpha=0.1))
    model_m.add(MaxPooling1D(2, strides=2))
    model_m.add(Dense (50))
    model_m.add(Dense (20))
    model_m.add(Flatten())
    model_m.add(Dense ( 3,activation='softmax'))
    print(model_m.summary())
    return model_m

dataset = numpy.loadtxt("newda.csv",delimiter=",")
X = dataset[:,0:177]
y = dataset[:,178]

# Instantiate the cross validator
skf = StratifiedKFold(n_splits=10, shuffle=True)
# Loop through the indices the split() method returns

for index, (train_indices, val_indices) in enumerate(skf.split(X, y)):
    print ("Training on fold   "+str(index+1)+"/10...")
    # Generate batches from indices
    xtrain, xval = X[train_indices], X[val_indices]
    ytrain, yval = y[train_indices], y[val_indices]
    
    # Clear model, and create it
    model = None
    model = create_model(xtrain,ytrain,xval,yval)
    
    # Debug message I guess
    print ("Training new iteration on " + str(xtrain.shape[0]) + " training samples, " + str(xval.shape[0]) + " validation samples, this may be a while...")
    # Compile model
    model.compile(loss='binary_crossentropy',optimizer='adam',metrics=['accuracy'],sample_weight_mode='temporal')
    
    xtrain = np.reshape(xtrain,(xtrain.shape[0], xtrain.shape[1],1))
    
    #ytrain = np.reshape(ytrain,(ytrain.shape[0], ytrain.shape[1],1))
    
    print(xtrain.shape, ytrain.shape, xval.shape, yval.shape)

    # Fit the model
    model.fit(xtrain,ytrain,epochs=15, batch_size=3, verbose=0)
    # evaluate the model
    scores = model.evaluate(xval, yval, verbose=0)
    print("%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))
    cvscores.append(scores[1] * 100)

**Comment by loda717** [comment]
hello ):
